app/Http/Controllers/SalonController.php

```diff
@@ -125,7 +125,65 @@ public function update(Request $request, Salon $salon)
      */
     public function destroy(Salon $salon)
     {
-        //
+        try {
+            DB::transaction(
+                function () use ($salon) {
+                    $salon = Salon::with(
+                        ['users',
+                        'categories',
+                        'customers.orders.bill',
+                        'customers.orders.products',
+                        ]
+                    )->find($salon->id);
+                    if ($salon != null) {
+                        $users = $salon->users;
+                        if ($users != null) {
+                            foreach ($users as $user) {
+                                $user->salons()->detach($salon->id);
+                            }
+                        }
```

Comment:
```suggestion
                        DB::table('salon_user')->where('salon_id', $id)->whereIn('user_id', $users->pluck('id'))->delete();
```
giảm số queries từ n -> 1... hoặc dùng cascade.